### PR TITLE
[bitnami/mongodb] Add optional parameters to configure service names used for mongo and arbiter services

### PR DESCRIPTION
**Pull request by jackieskeate**
**Description of the change**

Adds optional parameters to configure the service names used for the mongo and arbiter services.
If these optional parameters are not configured, the existing default service names will still be used to retain backwards compatibility.

**Benefits**

One use case is when moving to the bitnami chart from a different mongodb helm chart. This change means the service names can be configured so they remain the same as they were when using the previous helm chart.

**Possible drawbacks**

No known limitations.

**Applicable issues**

No applicable issues - this is not a bugfix.

**Additional information**

**Checklist** 
- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [X] Variables are documented in the README.md
- [X] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)

:warning: Keep in mind that if you want to make changes to the kubeapps chart, please implement them in the [kubeapps repository](https://github.com/kubeapps/kubeapps/tree/master/chart/kubeapps). This is only a synchronized mirror.

### CODE REVIEW

File: bitnami/mongodb/Chart.yaml [hunk 1]
```diff
@@ -26,4 +26,5 @@ name: mongodb
 sources:
   - https://github.com/bitnami/bitnami-docker-mongodb
   - https://mongodb.org
-version: 10.6.1
+version: 10.6.2
+
```
Comment (Mauraza) [inline bitnami/mongodb/Chart.yaml:30]: could you delete this blank line?

File: bitnami/mongodb/README.md [hunk 2]
```diff
@@ -220,6 +220,7 @@ The following tables lists the configurable parameters of the MongoDB chart and
 | Parameter                                         | Description                                                                                       | Default                        |
 |---------------------------------------------------|---------------------------------------------------------------------------------------------------|--------------------------------|
 | `service.type`                                    | Kubernetes Service type                                                                           | `ClusterIP`                    |
+| `service.name`                                    | MongoDB service name                                                                              | `{mongodb.fullname}-headless`  |
```
Comment (Mauraza) [inline bitnami/mongodb/README.md:223]: Could you rename service.name to service.nameOverride?
We used fullnameOVerridde and nameOverride, in this wat will be the same format.

File: bitnami/mongodb/README.md [hunk 3]
```diff
@@ -285,46 +286,47 @@ The following tables lists the configurable parameters of the MongoDB chart and
...
+| `arbiter.pdb.maxUnavailable`        | Maximum number/percentage of Arbiter pods that may be made unavailable                            | `nil`                                  |
+| `arbiter.initContainers`            | Add additional init containers for the Arbiter pod(s)                                             | `{}` (evaluated as a template)         |
+| `arbiter.sidecars`                  | Add additional sidecar containers for the Arbiter pod(s)                                          | `{}` (evaluated as a template)         |
+| `arbiter.extraVolumeMounts`         | Optionally specify extra list of additional volumeMounts for the Arbiter container(s)             | `{}`                                   |
+| `arbiter.extraVolumes`              | Optionally specify extra list of additional volumes to the Arbiter statefulset                    | `{}`                                   |
+| `arbiter.service.name`              | The arbiter service name                                                                          | `{mongodb.fullname}-arbiter-headless`  |
```
Comment (Mauraza) [inline bitnami/mongodb/README.md:329]: Change arbiter.service.name to arbiter.service.nameOverride

File: bitnami/mongodb/templates/_helpers.tpl [hunk 4]
```diff
@@ -15,6 +15,36 @@ If release name contains chart name it will be used as a full name.
 {{- include "common.names.fullname" . -}}
 {{- end -}}
 
+{{/*
+Create a default mongo service name which can be overridden.
+*/}}
+{{- define "mongodb.servicename" -}}
+    {{- if .Values.service -}}
+        {{- if .Values.service.name }}
```
Comment (Mauraza) [inline bitnami/mongodb/templates/_helpers.tpl:23]: Change the template to use service.nameOverride instead servicename

File: bitnami/mongodb/templates/_helpers.tpl [hunk 5]
```diff
@@ -15,6 +15,36 @@ If release name contains chart name it will be used as a full name.
...
+        {{ include "mongodb.fullname" . }}-headless
+    {{- end }}
+{{- end }}
+
+{{/*
+Create a default mongo arbiter service name which can be overridden.
+*/}}
+{{- define "mongodb.arbiterservicename" -}}
+    {{- if .Values.arbiter.service -}}
+        {{- if .Values.arbiter.service.name }}
```
Comment (Mauraza) [inline bitnami/mongodb/templates/_helpers.tpl:38]: Change servicename to service.nameOverride

File: bitnami/mongodb/templates/arbiter/headless-svc.yaml [hunk 6]
```diff
@@ -2,7 +2,7 @@
 apiVersion: v1
 kind: Service
 metadata:
-  name: {{ include "mongodb.fullname" . }}-arbiter-headless
+  name: {{ include "mongodb.arbiterservicename" . }}
```
Comment (Mauraza) [inline bitnami/mongodb/templates/arbiter/headless-svc.yaml:5]: Change arbiterservicename to arbiter.service.nameOverride

File: bitnami/mongodb/templates/arbiter/statefulset.yaml [hunk 7]
```diff
@@ -13,7 +13,7 @@ metadata:
   annotations: {{- include "common.tplvalues.render" (dict "value" .Values.arbiter.annotations "context" $) | nindent 4 }}
   {{- end }}
 spec:
-  serviceName: {{ include "mongodb.fullname" . }}-arbiter-headless
+  serviceName: {{ include "mongodb.arbiterservicename" . }}
```
Comment (Mauraza) [inline bitnami/mongodb/templates/arbiter/statefulset.yaml:16]: Change arbiterservicename to arbiter.service.nameOverride

File: bitnami/mongodb/templates/arbiter/statefulset.yaml [hunk 8]
```diff
@@ -145,11 +145,11 @@ spec:
                 fieldRef:
                   fieldPath: metadata.namespace
             - name: K8S_SERVICE_NAME
-              value: "{{ include "mongodb.fullname" . }}-arbiter-headless"
+              value: "{{ include "mongodb.arbiterservicename" . }}"
```
Comment (Mauraza) [inline bitnami/mongodb/templates/arbiter/statefulset.yaml:148]: Change arbiterservicename to arbiter.service.nameOverride

File: bitnami/mongodb/templates/arbiter/statefulset.yaml [hunk 9]
```diff
@@ -145,11 +145,11 @@ spec:
...
             - name: MONGODB_REPLICA_SET_MODE
               value: "arbiter"
             - name: MONGODB_INITIAL_PRIMARY_HOST
-              value: "{{ include "mongodb.fullname" . }}-0.{{ include "mongodb.fullname" . }}-headless.$(MY_POD_NAMESPACE).svc.{{ .Values.clusterDomain }}"
+              value: "{{ include "mongodb.fullname" . }}-0.{{ include "mongodb.servicename" . }}.$(MY_POD_NAMESPACE).svc.{{ .Values.clusterDomain }}"
```
Comment (Mauraza) [inline bitnami/mongodb/templates/arbiter/statefulset.yaml:152]: change servicename to servicenameOverride

File: bitnami/mongodb/templates/replicaset/headless-svc.yaml [hunk 10]
```diff
@@ -2,7 +2,7 @@
 apiVersion: v1
 kind: Service
 metadata:
-  name: {{ include "mongodb.fullname" . }}-headless
+  name: {{ include "mongodb.servicename" . }}
```
Comment (Mauraza) [inline bitnami/mongodb/templates/replicaset/headless-svc.yaml:5]: change servicename to servicenameOverride

File: bitnami/mongodb/templates/replicaset/statefulset.yaml [hunk 11]
```diff
@@ -16,7 +16,7 @@ metadata:
   annotations: {{- include "common.tplvalues.render" (dict "value" .Values.annotations "context" $) | nindent 4 }}
   {{- end }}
 spec:
-  serviceName: {{ include "mongodb.fullname" . }}-headless
+  serviceName: {{ include "mongodb.servicename" . }}
```
Comment (Mauraza) [inline bitnami/mongodb/templates/replicaset/statefulset.yaml:19]: change servicename to servicenameOverride

File: bitnami/mongodb/templates/replicaset/statefulset.yaml [hunk 12]
```diff
@@ -215,7 +215,7 @@ spec:
                 fieldRef:
                   fieldPath: metadata.namespace
             - name: K8S_SERVICE_NAME
-              value: "{{ include "mongodb.fullname" . }}-headless"
+              value: "{{ include "mongodb.servicename" . }}"
```
Comment (Mauraza) [inline bitnami/mongodb/templates/replicaset/statefulset.yaml:218]: change servicename to servicenameOverride

File: bitnami/mongodb/values.yaml [hunk 13]
```diff
@@ -483,6 +483,9 @@ persistence:
 ## Service parameters
 ##
 service:
+  ## Service name
+  ##
+  name: ""
   ## Service type
```
Comment (Mauraza) [inline bitnami/mongodb/values.yaml:489]: use nameOverride

File: bitnami/mongodb/values.yaml [hunk 14]
```diff
@@ -839,6 +842,12 @@ arbiter:
     ##
     # maxUnavailable: 1
 
+  ## MongoDB Arbiter service parameters
+  service:
+    ## Service name
+    ##
+    name: ""
+
 ## ServiceAccount
```
Comment (Mauraza) [inline bitnami/mongodb/values.yaml:851]: use nameOverride

**Review by Mauraza — COMMENTED** [review]
Thanks for the contribution!
I added some comments, a little change of the name.

File: bitnami/mongodb/Chart.yaml (hunk 1 above)
Comment (jackieskeate) [reply bitnami/mongodb/Chart.yaml]: Fixed

File: bitnami/mongodb/README.md (hunk 2 above)
Comment (jackieskeate) [reply bitnami/mongodb/README.md]: Fixed

File: bitnami/mongodb/README.md (hunk 3 above)
Comment (jackieskeate) [reply bitnami/mongodb/README.md]: Fixed

File: bitnami/mongodb/templates/_helpers.tpl (hunk 4 above)
Comment (jackieskeate) [reply bitnami/mongodb/templates/_helpers.tpl]: Fixed

File: bitnami/mongodb/templates/_helpers.tpl (hunk 5 above)
Comment (jackieskeate) [reply bitnami/mongodb/templates/_helpers.tpl]: Fixed

File: bitnami/mongodb/values.yaml (hunk 13 above)
Comment (jackieskeate) [reply bitnami/mongodb/values.yaml:489]: Fixed

File: bitnami/mongodb/values.yaml (hunk 14 above)
Comment (jackieskeate) [reply bitnami/mongodb/values.yaml:851]: Fixed

File: bitnami/mongodb/templates/arbiter/headless-svc.yaml (hunk 6 above)
Comment (jackieskeate) [reply bitnami/mongodb/templates/arbiter/headless-svc.yaml]: Fixed

File: bitnami/mongodb/templates/replicaset/statefulset.yaml (hunk 11 above)
Comment (jackieskeate) [reply bitnami/mongodb/templates/replicaset/statefulset.yaml]: Fixed

File: bitnami/mongodb/templates/replicaset/statefulset.yaml (hunk 12 above)
Comment (jackieskeate) [reply bitnami/mongodb/templates/replicaset/statefulset.yaml]: Fixed

File: bitnami/mongodb/templates/arbiter/statefulset.yaml (hunk 7 above)
Comment (jackieskeate) [reply bitnami/mongodb/templates/arbiter/statefulset.yaml]: Fixed

File: bitnami/mongodb/templates/arbiter/statefulset.yaml (hunk 8 above)
Comment (jackieskeate) [reply bitnami/mongodb/templates/arbiter/statefulset.yaml]: Fixed

File: bitnami/mongodb/templates/arbiter/statefulset.yaml (hunk 9 above)
Comment (jackieskeate) [reply bitnami/mongodb/templates/arbiter/statefulset.yaml]: Fixed

File: bitnami/mongodb/Chart.yaml (hunk 1 above)
Comment (jackieskeate) [reply bitnami/mongodb/Chart.yaml]: HI, thank you for all the feedback, I hope you are happy with these fixes, please let me know if you would like me to change anything else.

**Comment by Mauraza** [comment]
LGTM

**Pull request merged**